restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -554,6 +554,20 @@ else if (KeyUnstructuredDataResource.class.isAssignableFrom(collectionResourceCl
         @SuppressWarnings("unchecked")
         Class<? extends KeyUnstructuredDataResource<?>> clazz = (Class<? extends KeyUnstructuredDataResource<?>>) collectionResourceClass;
         actualTypeArguments = ReflectionUtils.getTypeArgumentsParametrized(KeyUnstructuredDataResource.class, clazz);
+        for (Method method : collectionResourceClass.getDeclaredMethods())
```

Comment:
This logic seems overly complicated to me, and it's also duplicated. Could you not simply have a check in `validateFinderMethod` which accesses the class type via the `ResourceModel`? Perhaps I'm missing something...

Reply:
Consolidated to the method 'validateFinderMethod'